index.html

```diff
@@ -834,22 +834,29 @@ <h2>
         |screenOrientation|'s {{ScreenOrientation/[[angle]]}}, abort these
         steps.
         </li>
-        <li>Set |screenOrientation|'s {{ScreenOrientation/[[angle]]}} to
-        |angle|.
-        </li>
-        <li>Set |screenOrientation|'s {{ScreenOrientation/[[type]]}} to |type|.
-        </li>
-        <li>[=Fire an event=] named "<a data-link-for=
-        "ScreenOrientation">change</a>" at |screenOrientation|.
+        <li>[=Queue a task=] on the [=user interaction task source=] with
+        |document|'s [=relevant global object=] to perform the following
+        substeps:
+          <ol>
+            <li>Set |screenOrientation|'s {{ScreenOrientation/[[angle]]}} to
+            |angle|.
+            </li>
+            <li>Set |screenOrientation|'s {{ScreenOrientation/[[type]]}} to
+            |type|.
+            </li>
+            <li>[=Fire an event=] named "<a data-link-for=
+            "ScreenOrientation">change</a>" at |screenOrientation|.
+            </li>
+          </ol>
         </li>
         <li>Let |descendantDocs| be an [=ordered set=] consisting of
         |document|'s [=Document/descendant navigables=]'s [=navigable/active
         documents=], if any, in tree order.
         </li>
         <li>[=Set/For each=] |doc:Document| in |descendantDocs|, [=queue a
-        global task=] on the [=DOM manipulation task source=] with |doc|'s
-        [=relevant global object=] to run the [=screen orientation change
-        steps=] with |doc|.
+        task=] on the [=user interaction task source=] with |doc|'s [=relevant
+        global object=] to run the [=screen orientation change steps=] with
+        |doc|.
```

Comment:
This seems like it's queueing a task to queue a task. That seems bad? Perhaps "screen orientation change steps" shouldn't queue a task itself after all?

Reply:
yeah, you are right... will just iterate over the docs.